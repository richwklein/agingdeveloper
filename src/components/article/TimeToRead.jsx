import React from "react";
import {Box, Card, Stack, Typography} from "@mui/material";
import PropTypes from "prop-types";
import AccessTimeOutlinedIcon from "@mui/icons-material/AccessTimeOutlined";

const TimeToRead = ({minutes, words, lang}) => {
  const wordFormat = new Intl.NumberFormat(lang).format(words);
  const readTemplate = `${Math.ceil(minutes)} min read`;
  const wordsTemplate = `${wordFormat} words`;

  return (
    <Card variant="outlined" sx={{padding: .5}}>
      <Stack direction="row" useFlexGap alignItems="center" spacing={1}>
        <AccessTimeOutlinedIcon />
        <Box flex={1}>
          <Typography component={Box} variant="body2">
            {readTemplate}
          </Typography>
          <Typography component={Box} variant="body2">
            {wordsTemplate}
          </Typography>
        </Box>
      </Stack>
    </Card>
  );
};

TimeToRead.propTypes = {
  minutes: PropTypes.number,
  words: PropTypes.number,
  lang: PropTypes.string.isRequired,
};

export default TimeToRead;
