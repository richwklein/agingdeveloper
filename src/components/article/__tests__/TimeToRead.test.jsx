import React from "react";
import {cleanup, render} from "@testing-library/react";
import TimeToRead from "../TimeToRead";

describe("TimeToRead", () => {
  afterEach(cleanup);

  const renderComponent = () => render(
      <TimeToRead
        minutes={12.25}
        words={99999}
        lang="en-us" />,
  );

  test("that it renders correctly.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("that minutes get rounded up.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toContainHTML("13 min read");
  });

  test("that word count is formatted based on lang.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toContainHTML("99,999 words");
  });
});
