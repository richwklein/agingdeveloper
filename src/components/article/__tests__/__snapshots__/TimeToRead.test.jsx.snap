// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeToRead that it renders correctly. 1`] = `
<div
  class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1hqzemx-MuiPaper-root-MuiCard-root"
>
  <div
    class="MuiStack-root css-1qxw92a-MuiStack-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="AccessTimeOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z"
      />
    </svg>
    <div
      class="MuiBox-root css-1rr4qq7"
    >
      <div
        class="MuiTypography-root MuiTypography-body2 MuiBox-root css-pc18se-MuiTypography-root"
      >
        13 min read
      </div>
      <div
        class="MuiTypography-root MuiTypography-body2 MuiBox-root css-pc18se-MuiTypography-root"
      >
        99,999 words
      </div>
    </div>
  </div>
</div>
`;
